Heads up that the current head (`c0652cd`) doesn't compile, so that needs sorting before the rest can be tested. Running `go build` / `gofmt -e` on internal/service/bedrockagentcore/memory_strategy.go there are four issues, two syntax and two undefined symbols.

1. Extra closing brace in `ValidateConfig`:
```
internal/service/bedrockagentcore/memory_strategy.go:301:1: expected declaration, found '}'
```
The `else` branch closes at line 299 and then there are two `}` (lines 300 and 301), one too many for the function body. Dropping the stray `}` at line 300 fixes it.

2. The new `if t.Reflection != nil` is mis-nested inside `if t.Extraction != nil` in `customConfigurationModel.Flatten`:
```
internal/service/bedrockagentcore/memory_strategy.go:885:2: expected '}', found 'default'
```
The `if t.Reflection != nil` (line 864) opens while still inside the `if t.Extraction != nil` block (line 850), and the resulting extra `}` at line 884 closes `switch t := v.(type)` before its `default:` case (line 885). The fix is to close the extraction block before the reflection handling and make `if t.Reflection` a sibling of `if t.Extraction`. Worth calling out that this isn't only the brace, as written reflection only gets read back when the API returns a non-nil Extraction, so making it a sibling fixes that too.

3. The `reflection_configuration` block is in the Attributes map instead of Blocks. At line 91 `reflection_configuration: schema.ListNestedBlock{...}` is added to the top-level Attributes map, but `ListNestedBlock` is a Block not an Attribute so it won't type-check (`ListNestedBlock does not implement schema.Attribute`). The nested `reflection` block at line 166 is under the Blocks map correctly, this top-level one just needs to move to the resource's Blocks map. This one is masked until 1 and 2 are fixed since the build stops at the syntax errors first.

4. `fwtypes.ListOfStringValueFrom` doesn't exist. Both the Create flatten (line 366) and Read flatten (line 428) call it, but there's no `ListOfStringValueFrom` in internal/framework/types. The existing helper for []string to ListOfString is `fwflex.FlattenFrameworkStringValueListOfString` (internal/framework/flex/list.go:39), which returns a single value with no diags, so both call sites become:
```go
rc.NamespaceTemplates = fwflex.FlattenFrameworkStringValueListOfString(ctx, episodicReflection.Value.NamespaceTemplates)
```

Once those are in the package should build and the acceptance tests can actually run the new paths. Happy to test the reflection round-trip against real AWS once it compiles. Thanks!

--- a/.changelog/48758.txt
+++ b/.changelog/48758.txt
@@ -1,0 +1,3 @@
+```release-note:enhancement
+resource/aws_bedrockagentcore_memory_strategy: Add `reflection_configuration` configuration block for `EPISODIC` strategy type
+```

--- a/internal/service/bedrockagentcore/memory_strategy.go
+++ b/internal/service/bedrockagentcore/memory_strategy.go
@@ -88,6 +88,15 @@ func (r *resourceMemoryStrategy) Schema(ctx context.Context, request resource.Sc
 				CustomType: fwtypes.SetOfStringType,
 				Required:   true,
 			},
+			names.AttrType: schema.StringAttribute{
+				Required:   true,
+				CustomType: fwtypes.StringEnumType[awstypes.MemoryStrategyType](),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
 			"reflection_configuration": schema.ListNestedBlock{
 				CustomType: fwtypes.NewListNestedObjectTypeOf[reflectionConfigurationModel](ctx),
 				Validators: []validator.List{
@@ -102,15 +111,6 @@ func (r *resourceMemoryStrategy) Schema(ctx context.Context, request resource.Sc
 					},
 				},
 			},
-			names.AttrType: schema.StringAttribute{
-				Required:   true,
-				CustomType: fwtypes.StringEnumType[awstypes.MemoryStrategyType](),
-				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.RequiresReplace(),
-				},
-			},
-		},
-		Blocks: map[string]schema.Block{
 			names.AttrConfiguration: schema.ListNestedBlock{
 				CustomType: fwtypes.NewListNestedObjectTypeOf[customConfigurationModel](ctx),
 				Validators: []validator.List{
@@ -163,29 +163,29 @@ func (r *resourceMemoryStrategy) Schema(ctx context.Context, request resource.Sc
 								},
 							},
 						},
-							"reflection": schema.ListNestedBlock{
-								CustomType: fwtypes.NewListNestedObjectTypeOf[reflectionOverrideModel](ctx),
-								Validators: []validator.List{
-									listvalidator.SizeAtMost(1),
-								},
-								PlanModifiers: []planmodifier.List{
-									errorIfSingleBlockRemoved("reflection"),
-								},
-								NestedObject: schema.NestedBlockObject{
-									Attributes: map[string]schema.Attribute{
-										"append_to_prompt": schema.StringAttribute{
-											Required: true,
-										},
-										"model_id": schema.StringAttribute{
-											Required: true,
-										},
-										"namespace_templates": schema.ListAttribute{
-											CustomType: fwtypes.ListOfStringType,
-											Optional:   true,
-										},
+						"reflection": schema.ListNestedBlock{
+							CustomType: fwtypes.NewListNestedObjectTypeOf[reflectionOverrideModel](ctx),
+							Validators: []validator.List{
+								listvalidator.SizeAtMost(1),
+							},
+							PlanModifiers: []planmodifier.List{
+								errorIfSingleBlockRemoved("reflection"),
+							},
+							NestedObject: schema.NestedBlockObject{
+								Attributes: map[string]schema.Attribute{
+									"append_to_prompt": schema.StringAttribute{
+										Required: true,
+									},
+									"model_id": schema.StringAttribute{
+										Required: true,
+									},
+									"namespace_templates": schema.ListAttribute{
+										CustomType: fwtypes.ListOfStringType,
+										Optional:   true,
 									},
 								},
 							},
+						},
 					},
 				},
 			},
@@ -297,7 +297,6 @@ func (r *resourceMemoryStrategy) ValidateConfig(ctx context.Context, request res
 			}
 		}
 	}
-	}
 }
 
 func (r *resourceMemoryStrategy) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
@@ -363,7 +362,7 @@ func (r *resourceMemoryStrategy) Create(ctx context.Context, request resource.Cr
 			if episodicReflection, ok := found.Configuration.Reflection.(*awstypes.ReflectionConfigurationMemberEpisodicReflectionConfiguration); ok {
 				var rc reflectionConfigurationModel
 				if episodicReflection.Value.NamespaceTemplates != nil {
-					rc.NamespaceTemplates, _ = fwtypes.ListOfStringValueFrom(ctx, episodicReflection.Value.NamespaceTemplates)
+					rc.NamespaceTemplates = fwflex.FlattenFrameworkStringValueListOfString(ctx, episodicReflection.Value.NamespaceTemplates)
 				}
 				plan.ReflectionConfiguration, _ = fwtypes.NewListNestedObjectValueOfPtr(ctx, &rc)
 			}
@@ -425,7 +424,7 @@ func (r *resourceMemoryStrategy) Read(ctx context.Context, request resource.Read
 		if episodicReflection, ok := out.Configuration.Reflection.(*awstypes.ReflectionConfigurationMemberEpisodicReflectionConfiguration); ok {
 			var rc reflectionConfigurationModel
 			if episodicReflection.Value.NamespaceTemplates != nil {
-				rc.NamespaceTemplates, _ = fwtypes.ListOfStringValueFrom(ctx, episodicReflection.Value.NamespaceTemplates)
+				rc.NamespaceTemplates = fwflex.FlattenFrameworkStringValueListOfString(ctx, episodicReflection.Value.NamespaceTemplates)
 			}
 			state.ReflectionConfiguration, _ = fwtypes.NewListNestedObjectValueOfPtr(ctx, &rc)
 		}
@@ -681,16 +680,16 @@ func findMemoryStrategyByTwoPartKey(ctx context.Context, conn *bedrockagentcorec
 
 type memoryStrategyResourceModel struct {
 	framework.WithRegionModel
-	Configuration          fwtypes.ListNestedObjectValueOf[customConfigurationModel] `tfsdk:"configuration"`
-	Description            types.String                                              `tfsdk:"description"`
-	MemoryExecutionRoleARN fwtypes.ARN                                               `tfsdk:"memory_execution_role_arn"`
-	MemoryStrategyID       types.String                                              `tfsdk:"memory_strategy_id"`
-	MemoryID               types.String                                              `tfsdk:"memory_id"`
-	Name                   types.String                                              `tfsdk:"name"`
-	Namespaces             fwtypes.SetOfString                                       `tfsdk:"namespaces"`
-	ReflectionConfiguration  fwtypes.ListNestedObjectValueOf[reflectionConfigurationModel] `tfsdk:"reflection_configuration"`
-	Type                   fwtypes.StringEnum[awstypes.MemoryStrategyType]           `tfsdk:"type"`
-	Timeouts               timeouts.Value                                            `tfsdk:"timeouts"`
+	Configuration           fwtypes.ListNestedObjectValueOf[customConfigurationModel]     `tfsdk:"configuration"`
+	Description             types.String                                                  `tfsdk:"description"`
+	MemoryExecutionRoleARN  fwtypes.ARN                                                   `tfsdk:"memory_execution_role_arn"`
+	MemoryStrategyID        types.String                                                  `tfsdk:"memory_strategy_id"`
+	MemoryID                types.String                                                  `tfsdk:"memory_id"`
+	Name                    types.String                                                  `tfsdk:"name"`
+	Namespaces              fwtypes.SetOfString                                           `tfsdk:"namespaces"`
+	ReflectionConfiguration fwtypes.ListNestedObjectValueOf[reflectionConfigurationModel] `tfsdk:"reflection_configuration"`
+	Type                    fwtypes.StringEnum[awstypes.MemoryStrategyType]               `tfsdk:"type"`
+	Timeouts                timeouts.Value                                                `tfsdk:"timeouts"`
 }
 
 func (m *memoryStrategyResourceModel) GetIdentifier() string {
@@ -815,10 +814,10 @@ func (m memoryStrategyResourceModel) expandToModifyMemoryStrategyInput(ctx conte
 }
 
 type customConfigurationModel struct {
-	Type          fwtypes.StringEnum[awstypes.OverrideType]             `tfsdk:"type"`
-	Consolidation fwtypes.ListNestedObjectValueOf[overrideDetailsModel] `tfsdk:"consolidation"`
-	Extraction    fwtypes.ListNestedObjectValueOf[overrideDetailsModel] `tfsdk:"extraction"`
-	Reflection    fwtypes.ListNestedObjectValueOf[reflectionOverrideModel]   `tfsdk:"reflection"`
+	Type          fwtypes.StringEnum[awstypes.OverrideType]                `tfsdk:"type"`
+	Consolidation fwtypes.ListNestedObjectValueOf[overrideDetailsModel]    `tfsdk:"consolidation"`
+	Extraction    fwtypes.ListNestedObjectValueOf[overrideDetailsModel]    `tfsdk:"extraction"`
+	Reflection    fwtypes.ListNestedObjectValueOf[reflectionOverrideModel] `tfsdk:"reflection"`
 }
 
 var (
@@ -860,6 +859,7 @@ func (m *customConfigurationModel) Flatten(ctx context.Context, v any) (diags di
 					return diags
 				}
 			}
+		}
 
 		if t.Reflection != nil {
 			switch rt := t.Reflection.(type) {
@@ -879,8 +879,6 @@ func (m *customConfigurationModel) Flatten(ctx context.Context, v any) (diags di
 					}
 				}
 			}
-		}
-		}
 		}
 	default:
 		diags.AddError(
@@ -972,7 +970,6 @@ func (m customConfigurationModel) expandToModifyStrategyConfiguration(ctx contex
 			return nil, diags
 		}
 	}
-
 
 	var reflection *reflectionOverrideModel
 	if !m.Reflection.IsNull() {

